pkg/webhook/resources/schedulevmbackup/validator.go

```diff
@@ -0,0 +1,169 @@
+		return fmt.Errorf("setting %s is empty", settings.BackupTargetSettingName)
+	}
+
+	target, err := settings.DecodeBackupTarget(targetSetting.Value)
+	if err != nil {
+		return fmt.Errorf("can't decode setting %s value %s, error: %w", settings.BackupTargetSettingName, targetSetting.Value, err)
+	}
+
+	if target.IsDefaultBackupTarget() {
+		return fmt.Errorf("setting %s is not set", settings.BackupTargetSettingName)
+	}
+
+	if target.Type == settings.S3BackupType {
+		secret, err := v.secretCache.Get(util.LonghornSystemNamespaceName, util.BackupTargetSecretName)
+		if err != nil {
+			return fmt.Errorf("can't get backup target secret: %s/%s, error: %w",
+				util.LonghornSystemNamespaceName, util.BackupTargetSecretName, err)
+		}
+		os.Setenv(backup.AWSAccessKey, string(secret.Data[backup.AWSAccessKey]))
+		os.Setenv(backup.AWSSecretKey, string(secret.Data[backup.AWSSecretKey]))
+		os.Setenv(backup.AWSEndpoints, string(secret.Data[backup.AWSEndpoints]))
+		os.Setenv(backup.AWSCERT, string(secret.Data[backup.AWSCERT]))
+	}
+
+	_, err = backupstore.GetBackupStoreDriver(backup.ConstructEndpoint(target))
+	if err != nil {
+		return fmt.Errorf("can't connect to backup target %+v, error: %w", target, err)
+	}
+
+	return nil
+}
+
+func (v *scheuldeVMBackupValidator) Create(_ *types.Request, newObj runtime.Object) error {
+	newSVMBackup := newObj.(*v1beta1.ScheduleVMBackup)
+
+	if newSVMBackup.Spec.MaxFailure >= newSVMBackup.Spec.Retain {
+		return werror.NewInvalidError("max failure should be less than retain", fieldMaxFailure)
+	}
+
+	if _, err := cron.ParseStandard(newSVMBackup.Spec.Cron); err != nil {
+		return werror.NewInvalidError("invalid cron format", fieldCron)
+	}
+
+	srcVM := fmt.Sprintf("%s/%s", newSVMBackup.Namespace, newSVMBackup.Spec.VMBackupSpec.Source.Name)
+	svmbackups, err := v.svmbackupCache.GetByIndex(indexeres.ScheduleVMBackupBySourceVM, srcVM)
+	if err != nil {
+		return err
+	}
+
+	if len(svmbackups) != 0 {
+		return werror.NewInvalidError(fmt.Sprintf("VM %s already has backup schedule", srcVM), fieldVMBackup)
+	}
+
+	if newSVMBackup.Spec.VMBackupSpec.Type == v1beta1.Snapshot {
+		return nil
+	}
+
+	if err := v.checkTargetHealth(); err != nil {
+		return werror.NewInvalidError(err.Error(), fieldSuspend)
+	}
+
+	return nil
+}
+
+func (v *scheuldeVMBackupValidator) Update(_ *types.Request, oldObj runtime.Object, newObj runtime.Object) error {
+	newSVMBackup := newObj.(*v1beta1.ScheduleVMBackup)
+	oldSVMBackup := oldObj.(*v1beta1.ScheduleVMBackup)
+
+	if !newSVMBackup.DeletionTimestamp.IsZero() {
+		return nil
+	}
+
+	if newSVMBackup.Spec.MaxFailure >= newSVMBackup.Spec.Retain {
+		return werror.NewInvalidError("max failure should be less than retain", fieldMaxFailure)
+	}
+
+	if _, err := cron.ParseStandard(newSVMBackup.Spec.Cron); err != nil {
+		return werror.NewInvalidError("invalid cron format", fieldCron)
+	}
+
+	//not updated to resume schedule
+	if !oldSVMBackup.Spec.Suspend || newSVMBackup.Spec.Suspend {
```

Comment:
Could you elaborate more on this condition? Thanks.

Reply:
If the old svmbackup CR is already in a resume state, the Harvester controller doesn't need to check backup target connectivity.

If the new svmbackup CR asks to suspend the schedule, the Harvester controller doesn't need to check backup target connectivity, either.

Thanks.